test/cmp_client_test.c

```diff
@@ -321,7 +324,8 @@ static int test_exchange_certConf(void)
 {
     SETUP_TEST_FIXTURE(CMP_SES_TEST_FIXTURE, set_up);
     fixture->expected = 0; /* client should not send certConf immediately */
-    if (!ossl_cmp_ctx_set0_newCert(fixture->cmp_ctx, X509_dup(client_cert))) {
+    if (!ossl_cmp_ctx_set0_newCert(fixture->cmp_ctx,
+                                   x509_dup_with_libctx(client_cert))) {
```

Comment:
As mentioned above - I think X509_dup should just do the right thing.

Reply:
Yep; removed.